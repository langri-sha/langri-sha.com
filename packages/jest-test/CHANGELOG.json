{
  "name": "@langri-sha/jest-test",
  "entries": [
    {
      "date": "Wed, 24 Apr 2024 22:03:36 GMT",
      "version": "0.2.0",
      "tag": "@langri-sha/jest-test_v0.2.0",
      "comments": {
        "patch": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/jest-test",
            "commit": "8d67a370e43a304d96b4e84a0eb773a9f7d4fecd",
            "comment": "feat(workspace): Use PNPM local references"
          }
        ],
        "minor": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/jest-test",
            "commit": "f9dcfcb1ce5b663131c076f5364425c03f757710",
            "comment": "feat(workspace): Update package metadata"
          }
        ]
      }
    }
  ]
}
