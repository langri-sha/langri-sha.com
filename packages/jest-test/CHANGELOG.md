# Change Log - @langri-sha/jest-test

This log was last generated on Fri, 03 May 2024 12:54:50 GMT and should not be manually modified.

<!-- Start content -->

## 0.7.1

Fri, 03 May 2024 12:54:50 GMT

### Patches

- fix(workspace): Correct published paths (filip.dupanovic@gmail.com)

## 0.7.0

Fri, 03 May 2024 10:44:32 GMT

### Minor changes

- feat(jest-test): Configure package build (filip.dupanovic@gmail.com)

### Patches

- fix(workspace): Sync packages (filip.dupanovic@gmail.com)
- fix(workspace): Revise build command (filip.dupanovic@gmail.com)
- fix(workspace): Sync packages (filip.dupanovic@gmail.com)

## 0.2.0

Wed, 24 Apr 2024 22:03:36 GMT

### Minor changes

- feat(workspace): Update package metadata (filip.dupanovic@gmail.com)

### Patches

- feat(workspace): Use PNPM local references (filip.dupanovic@gmail.com)
