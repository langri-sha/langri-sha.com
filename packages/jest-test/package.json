{
  "name": "@langri-sha/jest-test",
  "version": "0.1.0",
  "license": "MIT",
  "main": "src/index.ts",
  "dependencies": {
    "@jest/globals": "29.7.0",
    "tempy": "1.0.1"
  },
  "devDependencies": {
    "@langri-sha/tsconfig": "0.1.0"
  },
  "peerDependencies": {
    "jest": "^29.0.0",
    "typescript": "^5.0.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
