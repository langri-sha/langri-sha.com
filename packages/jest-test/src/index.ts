export * from '@jest/globals'
export { default as tempy } from 'tempy'
export { default as nock } from 'nock'
