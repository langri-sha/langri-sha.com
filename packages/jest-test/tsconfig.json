{
  "$schema": "https://json.schemastore.org/tsconfig",
  "extends": "@langri-sha/tsconfig/project.json",
  "references": [{ "path": "../tsconfig" }]
}
