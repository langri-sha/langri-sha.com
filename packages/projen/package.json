{
  "name": "@langri-sha/projen",
  "version": "0.1.0",
  "license": "MIT",
  "main": "src/index.ts",
  "devDependencies": {
    "@langri-sha/tsconfig": "workspace:*",
    "projen": "0.81.1"
  },
  "peerDependencies": {
    "projen": "^0.81.1",
    "typescript": "^5.0.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
