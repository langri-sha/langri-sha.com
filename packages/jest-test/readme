# @langri-sha/jest-test

Provides some useful helpers that are commonly used for authoring tests.

## Usage

Install the required dependencies:

```sh
npm install -D jest @langri-sha/jest-test
```

Then import your Jest test dependencies from here:

```js
// some.test.js
import { expect, test, tempy } from '@langri-sha/jest-test'

test(/*...*/)
```

## See

- [`@jest/globals`]
- [`tempy`]

[`@jest/globals`]: https://jestjs.io/docs/api
[`tempy`]: https://github.com/sindresorhus/tempy
