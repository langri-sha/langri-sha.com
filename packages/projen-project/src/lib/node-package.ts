import * as fs from 'node:fs'

import { Project, javascript } from 'projen'

/**
 * Options for configuring Node.js packages.
 */
export interface NodePackageOptions extends javascript.NodePackageOptions {
  peerDependenciesMeta?: Record<string, { optional?: boolean }>
  type?: 'commonjs' | 'module'
}

/**
 * Custom Node.js package manager that preserves package.
 */
export class NodePackage extends javascript.NodePackage {
  constructor(project: Project, options?: NodePackageOptions) {
    super(project, options)

    this.addVersion(this.#getPackageJson()?.version ?? '0.0.0')

    if (options?.type) {
      this.addField('type', options.type)
    }

    if (options?.peerDependenciesMeta) {
      this.addField('peerDependenciesMeta', options.peerDependenciesMeta)
    }

    // Reset, so we can run `pnpx projen` on CI after Renovate upgrades.
    this.project.tasks
      .tryFind('install:ci')
      ?.reset(this.project.tasks.tryFind('install')?.steps[0]?.exec)
  }

  #getPackageJson() {
    try {
      return JSON.parse(fs.readFileSync(this.file.absolutePath, 'utf-8'))
    } catch (_) {
      return undefined
    }
  }

  override postSynthesize(): void {
    if (this.project.parent) {
      return
    }

    super.postSynthesize()
  }
}
