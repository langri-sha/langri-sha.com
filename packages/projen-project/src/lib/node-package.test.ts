import { promises as fs } from 'node:fs'
import * as path from 'node:path'
import { describe, expect, tempy, test } from '@langri-sha/jest-test'
import { synthSnapshot } from 'projen/lib/util/synth'

import { NodePackage } from './node-package'
import { Project } from 'projen'

describe('resolves version', () => {
  test('without a package set', () => {
    const project = new Project({
      name: 'test',
    })

    const pkg = new NodePackage(project)
    pkg.addVersion('1.0.0')

    expect(synthSnapshot(project)['package.json']).toMatchSnapshot()
  })

  test('with a package set', async () => {
    const outdir = tempy.directory()

    await fs.writeFile(
      path.join(outdir, 'package.json'),
      JSON.stringify({
        version: '1.0.0',
      }),
    )

    const project = new Project({
      name: 'test-project',
      outdir,
    })

    new NodePackage(project)

    expect(synthSnapshot(project)['package.json']).toMatchSnapshot()
  })
})
