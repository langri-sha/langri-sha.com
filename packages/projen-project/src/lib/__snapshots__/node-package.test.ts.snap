// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
{
  ".gitignore": "# ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
node_modules/
!/.gitattributes
!/.projen/tasks.json
!/.projen/deps.json
!/.projen/files.json
",
  ".projen/files.json": {
    "//": "~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".",
    "files": [
      ".gitattributes",
      ".gitignore",
      ".projen/deps.json",
      ".projen/files.json",
      ".projen/tasks.json",
    ],
  },
  ".projen/tasks.json": {
    "//": "~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".",
    "tasks": {
      "build": {
        "description": "Full release build",
        "name": "build",
        "steps": [
          {
            "spawn": "default",
          },
          {
            "spawn": "pre-compile",
          },
          {
            "spawn": "compile",
          },
          {
            "spawn": "post-compile",
          },
          {
            "spawn": "test",
          },
          {
            "spawn": "package",
          },
        ],
      },
      "compile": {
        "description": "Only compile",
        "name": "compile",
      },
      "default": {
        "description": "Synthesize project files",
        "name": "default",
      },
      "eject": {
        "description": "Remove projen from the project",
        "env": {
          "PROJEN_EJECTING": "true",
        },
        "name": "eject",
        "steps": [
          {
            "spawn": "default",
          },
        ],
      },
      "package": {
        "description": "Creates the distribution package",
        "name": "package",
      },
      "post-compile": {
        "description": "Runs after successful compilation",
        "name": "post-compile",
      },
      "pre-compile": {
        "description": "Prepare the project for compilation",
        "name": "pre-compile",
      },
      "test": {
        "description": "Run tests",
        "name": "test",
      },
    },
  },
  "project2/.gitignore": "# ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
node_modules/
!/.gitattributes
!/.projen/tasks.json
!/.projen/deps.json
!/.projen/files.json
!/package.json
",
  "project2/.projen/files.json": {
    "//": "~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".",
    "files": [
      ".gitattributes",
      ".gitignore",
      ".projen/deps.json",
      ".projen/files.json",
      ".projen/tasks.json",
    ],
  },
  "project2/.projen/tasks.json": {
    "//": "~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".",
    "tasks": {
      "build": {
        "description": "Full release build",
        "name": "build",
        "steps": [
          {
            "spawn": "pre-compile",
          },
          {
            "spawn": "compile",
          },
          {
            "spawn": "post-compile",
          },
          {
            "spawn": "test",
          },
          {
            "spawn": "package",
          },
        ],
      },
      "compile": {
        "description": "Only compile",
        "name": "compile",
      },
      "default": {
        "description": "Synthesize project files",
        "name": "default",
        "steps": [
          {
            "cwd": "..",
            "exec": "npx projen default",
          },
        ],
      },
      "install": {
        "description": "Install project dependencies and update lockfile (non-frozen)",
        "name": "install",
        "steps": [
          {
            "exec": "yarn install --check-files",
          },
        ],
      },
      "install:ci": {
        "description": "Install project dependencies using frozen lockfile",
        "name": "install:ci",
        "steps": [
          {
            "exec": "yarn install --check-files",
          },
        ],
      },
      "package": {
        "description": "Creates the distribution package",
        "name": "package",
      },
      "post-compile": {
        "description": "Runs after successful compilation",
        "name": "post-compile",
      },
      "pre-compile": {
        "description": "Prepare the project for compilation",
        "name": "pre-compile",
      },
      "test": {
        "description": "Run tests",
        "name": "test",
      },
    },
  },
  "project2/package.json": {
    "//": "~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".",
    "license": "Apache-2.0",
    "main": "lib/index.js",
    "name": "test",
    "scripts": {
      "build": "npx projen build",
      "compile": "npx projen compile",
      "default": "npx projen default",
      "package": "npx projen package",
      "post-compile": "npx projen post-compile",
      "pre-compile": "npx projen pre-compile",
      "test": "npx projen test",
    },
    "version": "0.0.0",
  },
}
`;

exports[`assigns \`peerDependenciesMeta\` field 1`] = `
{
  "//": "~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".",
  "devDependencies": {
    "prettier": "0.0.0",
  },
  "license": "Apache-2.0",
  "main": "lib/index.js",
  "name": "test",
  "peerDependencies": {
    "prettier": "*",
  },
  "peerDependenciesMeta": {
    "prettier": {
      "optional": true,
    },
  },
  "scripts": {
    "build": "npx projen build",
    "compile": "npx projen compile",
    "default": "npx projen default",
    "eject": "npx projen eject",
    "package": "npx projen package",
    "post-compile": "npx projen post-compile",
    "pre-compile": "npx projen pre-compile",
    "test": "npx projen test",
  },
  "version": "0.0.0",
}
`;

exports[`assigns \`type\` field 1`] = `
{
  "//": "~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".",
  "license": "Apache-2.0",
  "main": "lib/index.js",
  "name": "test",
  "scripts": {
    "build": "npx projen build",
    "compile": "npx projen compile",
    "default": "npx projen default",
    "eject": "npx projen eject",
    "package": "npx projen package",
    "post-compile": "npx projen post-compile",
    "pre-compile": "npx projen pre-compile",
    "test": "npx projen test",
  },
  "type": "module",
  "version": "0.0.0",
}
`;

exports[`resolves version with a package set 1`] = `
{
  "//": "~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".",
  "license": "Apache-2.0",
  "main": "lib/index.js",
  "name": "test-project",
  "scripts": {
    "build": "npx projen build",
    "compile": "npx projen compile",
    "default": "npx projen default",
    "eject": "npx projen eject",
    "package": "npx projen package",
    "post-compile": "npx projen post-compile",
    "pre-compile": "npx projen pre-compile",
    "test": "npx projen test",
  },
  "version": "1.0.0",
}
`;

exports[`resolves version without a package set 1`] = `
{
  "//": "~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".",
  "license": "Apache-2.0",
  "main": "lib/index.js",
  "name": "test",
  "scripts": {
    "build": "npx projen build",
    "compile": "npx projen compile",
    "default": "npx projen default",
    "eject": "npx projen eject",
    "package": "npx projen package",
    "post-compile": "npx projen post-compile",
    "pre-compile": "npx projen pre-compile",
    "test": "npx projen test",
  },
  "version": "1.0.0",
}
`;
